openssl-dynamic/src/main/java/io/netty/internal/tcnative/OcspCallbackStatus.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.internal.tcnative;
+
+import static io.netty.internal.tcnative.NativeStaticallyReferencedJniMethods.sslTlsExtErrAlertFatal;
+import static io.netty.internal.tcnative.NativeStaticallyReferencedJniMethods.sslTlsExtErrAlertWarning;
+import static io.netty.internal.tcnative.NativeStaticallyReferencedJniMethods.sslTlsExtErrNoAck;
+import static io.netty.internal.tcnative.NativeStaticallyReferencedJniMethods.sslTlsExtErrOk;
+
+/**
+ * An enumeration of various client and server side OCSP callback status codes.
+ *
+ * @see OcspCallback#callback(long)
+ */
+public final class OcspCallbackStatus {
+
+    private OcspCallbackStatus() {}
+
+    // ATTENTION: This class is purposely separate from the OcspCallback interface
+    // as JNI will otherwise fail to initialize due to a circular dependencies that
```

Comment:
> It created a circular dependency as soon as I started using NativeStaticallyReferencedJniMethods for pulling the values from JNI.

I see. We manually control the class load sequencing in `transport-native-epoll` [1] to avoid this issue. We should be doing the same in tcnative too ... thx for pointing this out I will add a PR to correct this.

[1] https://github.com/netty/netty/blob/4.1/transport-native-epoll/src/main/c/netty_epoll_native.c#L849